io/plugins/eu.esdihumboldt.hale.io.gml/src/eu/esdihumboldt/hale/io/gml/reader/internal/wfs/WfsBackedGmlInstanceCollection.java

```diff
@@ -225,10 +225,16 @@ public WfsBackedGmlInstanceCollection(LocatableInputSupplier<? extends InputStre
 			}
 		}
 
+		// for a query containing RESOLVEDEPTH we disable the pagination
+		if (primordialQueryParams.containsKey("RESOLVEDEPTH")) {
```

Comment:
After discussing the issue with @florianesser, we identified a potential solution: disable pagination if the parameters contain "RESOLVEDEPTH".

Another approach could be to disable pagination directly from the UI. I attempted this but haven't found a viable solution yet.

@stempler, your input and ideas would be greatly appreciated to help us find the best solution.

Reply:
Problem with that would be that in many cases then it would not be possible to load all data of a service.

- Was your intent to inform the user about this restriction in the UI?
- Might be unexpected when used from another context, e.g. Gradle or CLI
- Did you explore if you would see some way to determine if an instance originated from `additionalObjects` or not? (e.g. see if it is feasible to detect [here](https://github.com/halestudio/hale/blob/d5e541ed65f91227e1c5cf1b10b89b7ebb777ba1/io/plugins/eu.esdihumboldt.hale.io.gml/src/eu/esdihumboldt/hale/io/gml/reader/internal/GmlInstanceCollection.java#L169) once entering `additionalObjects` and adding the information to created instances in the [instance metadata](https://github.com/halestudio/hale/blob/d5e541ed65f91227e1c5cf1b10b89b7ebb777ba1/common/plugins/eu.esdihumboldt.hale.common.instance/src/eu/esdihumboldt/hale/common/instance/model/Instance.java#L74))

Reply:
> Was your intent to inform the user about this restriction in the UI?

Initially, I wanted to disable the checkbox, but I didn't find the way, but if you have any suggestion how to achieve that I would be very happy to hear. 
1. I have tried to understand in the `GmlReaderSettingsPage` if there is any way to get the link and like that be able to check if the "resolvedepth" parameter is there.
2. I tried to extend  `GmlReaderSettingsPage` into the WFS UI package but without success. 
3. Not the best solution would be to add a note in the actual UI and inform the user that that parameter for steps won't be used in case "resolvedepth" has been used. 

> Might be unexpected when used from another context, e.g. Gradle or CLI

True. 

> Did you explore if you would see some way to determine if an instance originated from additionalObjects or not?

I have tried but again without getting the info where the instance comes from. I will definitely try your suggestion.